tests/test_advanced.py

```diff
@@ -108,6 +108,19 @@ def goo(*args, **kwargs):
     assert goo('hello') == 'hello'
 
 
+def tests_wraps_lambda():
+    """ Tests that `@wraps` can duplicate the signature of a lambda """
+    foo = lambda a: a
```

Comment:
Could you add a variant where `foo.__name__` is modified before calling `wraps` ? See above comment. Using `__code__.co_name` would fix the associated issue apparently.